nix-support-dev/lainchan.nix

```diff
@@ -4,8 +4,36 @@ let
   app = "lainchan";
   domain = "leftypol.org";
   dataDir = "/srv/http/${app}.leftypol.org";
-  #acmeRoot = "/var/lib/acme/acme-challenge";
   oldpkgs = import ./nixpkgs {};
+
+  leftypol_common_location_block = {
+    "~ \.php$" = {
+      root = dataDir;
+      extraConfig = ''
+            # fastcgi_split_path_info ^(.+\.php)(/.+)$;
+            fastcgi_pass unix:${config.services.phpfpm.pools.${app}.socket};
+      '';
+    };
+
+    "~* \.(?:html|json)$" = {
+      root = dataDir;
+      extraConfig = ''
+        expires 1s;
+      '';
+    };
+
+    "~* \.(jpg|jpeg|png|gif|ico|css|js|mp4|mp3|webm|pdf|bmp|zip|epub)$" = {
```

Comment:
Is this redundant?
If not, we should perhaps extend expiration to a very long time, since static images are unlikely to change.
If js files do change (we fix a file under `js/some-plugin.js`, we could always purge the cache if needed.
I'd say a week or so would be a good starting point.

Reply:
While we are rapidly making changes to the site still I don't want the user's browser to cache css and js for too long, so I set everything to 1h. You can purge the cloudflare cache but you can't force the user to clear theirs.

Reply:
I'll add an issue to remind us about increasing cache expiration.